org.eclipse.lsp4e/src/org/eclipse/lsp4e/LanguageServerWrapper.java

```diff
@@ -189,6 +193,8 @@ public void dirtyStateChanged(IFileBuffer buffer, boolean isDirty) {
 	private Timer timer;
 	private AtomicBoolean stopping = new AtomicBoolean(false);
 
+	private final ExecutorService dispatcher;
```

Comment:
Unfortunately I think that'd probably be of little use or even counterproductive: if people want the benefits/guarantees they need to use the new API so that the dispatcher is used *correctly*. If we expose the dispatcher then we risk it being abused in dependent plugins over which we have little control. Direct access would allow people to tie the thread up with long-running unrelated work or even cause a deadlock.

Reply:
ok